### DISCUSSION TITLE
Deleting what download files?

### DISCUSSION THREAD

**Discussion by jadaml** [moraroy/NonSteamLaunchers-On-Steam-Deck#527] [General]
Hi @moraroy ,
When I was reading the [README.md][2] file, I saw a mention, that your script will delete the downloaded files, (without where it would download, and how it handles the downloaded files) and the reader should not worry. That caused me to worry, so I decided to look up what is it that is being deleted. I [found the code that specifies the download folder][1], and it does seems that this will create a new folder under downloads, and [later your script does delete only that folder][3] in multiple instances. But I wonder about the `$(eval echo ~$user)` part of the code. I didn't find a variable `user` (lowercase) being defined anywhere in the [NonSteamLauncher.sh][4] script, and I admit I did not searched for any other variable definition in other files, and it is not a typical environment variable that is being defined. Also, you are both using `~` and a variable, who's name suggest it should be the username, so it would be `~deck` on SteamOS. There *is,* however `HOME` and `USER` (in all-caps) defined in the environment variables, which seems to be suitable for this purpose. Also there is the `logged-in-user` variable defined which may be what you wanted to use? Is this a typo? Have you intended to use another variable? Or maybe you have intended to define the `user` (lowercase) variable, and just forgot?

Further inspection does not reveal anything unwanted erroneously deleted, but again, admittedly, I only checked the [NonSteamLauncher.sh][4] file.

[1]: https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/blob/2d75e133a6c40d1d7ca36ea28ba8928521e3db5c/NonSteamLaunchers.sh#L33
[2]: https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/blob/2d75e133a6c40d1d7ca36ea28ba8928521e3db5c/README.md
[3]: https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/blob/2d75e133a6c40d1d7ca36ea28ba8928521e3db5c/NonSteamLaunchers.sh#L2506
[4]: https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/blob/2d75e133a6c40d1d7ca36ea28ba8928521e3db5c/NonSteamLaunchers.sh

**Comment by moraroy** [comment]
@jadaml I probably do have a typo somewhere. I was just echoing the user, I tried my best to mostly use logged_in_user as that's defined. But even given all of this is there an error you're encountering?

**Reply by jadaml** [reply]
It did install fine, and it has downloaded files in the correct `~/Downloads/...` folder, but that is due to the fact, that `user` variable was not defined anywhere.
